### PR TITLE
Fix the circle pattern board when it is vertically oriented

### PR DESCRIPTION
**Pull request by knzo25**
## PR Type

- Bug Fix

## Related Links

[Jira ticket](https://tier4.atlassian.net/browse/T4PB-17397)
[Data to validate the fix](https://drive.google.com/drive/folders/1OyZH9JXDXd5ooza7unLdy7WjKDyZiZE_?usp=sharing)

## Description
The circle pattern detector has to be run twice depending on the orientation of the board.
However, when the board was detected vertically, the points where not in the order the rest of the code expected causing a completely wrong camera calibration result.

For example, the focal distance should be around 1000 px, but the estimation could be from 3.000 px to 10.000 px

## Review Procedure

1. Perform camera calibration without this PR and the provided data and confirm the previously stated error.
2. Using the PR validate that the calibration provides correct camera intrinsics

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/KB4FAE/pages/1748435052
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: camera_calibration/src/camera_calibration/calibrator.py [hunk 1]
```diff
@@ -246,6 +246,11 @@ def _get_circles(img, board, pattern):
         (ok, corners) = cv2.findCirclesGrid(
             mono_arr, (board.n_rows, board.n_cols), flags=flag, blobDetector=detector)
 
+        if ok:
+            corners = corners.reshape((board.n_cols, board.n_rows, 2))
```
Comment (yabuta) [inline camera_calibration/src/camera_calibration/calibrator.py]: ここでcorners変数を使い回してますが後から別の人が読んだ時に直感的にわかりづらいかと思います
あまり可読性がよくないと思うので別々の変数名をつけるのが良いと思います

ex)
reshaped_corners = corners.reshape((board.n_cols, board.n_rows, 2))
transposed_corners = np.transpose(reshaped_corners, (1, 0, 2))
corners = transposed_corners.reshape(-1, 1, 2)

**Review by yabuta — DISMISSED** [review]

File: camera_calibration/src/camera_calibration/calibrator.py (hunk 1 above)
Comment (knzo25) [reply camera_calibration/src/camera_calibration/calibrator.py]: done !

File: camera_calibration/src/camera_calibration/calibrator.py [hunk 2]
```diff
@@ -246,6 +246,11 @@ def _get_circles(img, board, pattern):
         (ok, corners) = cv2.findCirclesGrid(
             mono_arr, (board.n_rows, board.n_cols), flags=flag, blobDetector=detector)
 
+        if ok:
```
Comment (aohsato) [inline camera_calibration/src/camera_calibration/calibrator.py:250]: [nits] add comments about your intention for this implementation.

**Review by aohsato — APPROVED** [review]
LGTM
nits x 1

File: camera_calibration/src/camera_calibration/calibrator.py (hunk 2 above)
Comment (knzo25) [reply camera_calibration/src/camera_calibration/calibrator.py:250]: done

**Pull request merged**